app/src/main/java/com/nervousfish/nervousfish/modules/database/DatabaseException.java

```diff
@@ -25,4 +29,30 @@ public DatabaseException(final Throwable throwable) {
         super(throwable);
     }
 
+    /**
+     * Deserialize the instance using readObject to ensure invariants and security.
+     *
+     * @param stream The serialized object to be deserialized
+     */
+    private void readObject(final ObjectInputStream stream) throws IOException, ClassNotFoundException {
```

Comment:
Hmm, I didn't generate it, talked with somebody from Forze about these kind of issues and he considered it good practice to explicitly declare objects that may be redundant at the time being but are not redundant anymore when class gets more functionality. When you add an attribute to a class you're much more likely to forget to add that not-so-obvious code that's needed as well. In this case I can imagine that when we add an attribute that describes the database file in which the error occurred, people will forget to add a Serialization Proxy to check if the database file is not initialized to null